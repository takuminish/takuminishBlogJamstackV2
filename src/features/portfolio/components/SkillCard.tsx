import Paper from "@mui/material/Paper";
import { Skill } from "../../../pages";
import Typography from "@mui/material/Typography";
import React from "react";

type Props = {
  skill: Skill;
};

function SkillCard(props: Props) {
  const { skill } = props;

  return (
    <Paper>
      {" "}
      <Typography variant="h6" sx={{ fontWeight: 700, textAlign: "center" }}>
        {skill.name}
      </Typography>
      <Typography variant="body1">
        <div
          dangerouslySetInnerHTML={{ __html: skill.detailedInfomation }}
        ></div>
      </Typography>
    </Paper>
  );
}

export default SkillCard;
