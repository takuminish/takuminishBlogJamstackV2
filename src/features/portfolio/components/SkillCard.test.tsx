import React from "react";
import { render } from "@testing-library/react";
import SkillCard, { Props } from "./SkillCard";

const createDOM = () => {
  const props: Props = {
    skill: {
      fieldId: "skill",
      name: "skill1",
      detailedInfomation: "<p>test</p>",
      yearOfExperience: 3,
      category: ["フレームワーク"],
    },
  };

  return render(<SkillCard skill={props.skill} />);
};

test("snapshot", () => {
  const { container } = createDOM();

  expect(container).toMatchSnapshot();
});
