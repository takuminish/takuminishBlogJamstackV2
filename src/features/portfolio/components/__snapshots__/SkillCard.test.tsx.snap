// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
  >
     
    <h6
      class="MuiTypography-root MuiTypography-h6 css-10v34n7-MuiTypography-root"
    >
      skill1
    </h6>
    <div>
      <p>
        test
      </p>
    </div>
  </div>
</div>
`;
